network/src/connectivity_manager/mod.rs

```diff
@@ -338,6 +340,26 @@ where
             })
             .collect();
 
+        // Limit the number of dialed connections from a Full Node
```

Comment:
i don't think this will work, since we'll probably keep trying to dial the same 3 peers even if we can't connect to them.

we should shuffle `to_connect` before slicing (can use https://docs.rs/rand/0.7.3/rand/seq/trait.SliceRandom.html#tymethod.shuffle or https://docs.rs/rand/0.7.3/rand/seq/trait.SliceRandom.html#tymethod.partial_shuffle if you're feeling fancy i guess)

you can setup a SmallRng like how protocol/discovery does it

Reply:
Good call! We absolutely do not want to be dialing the same 3 over and over again or across the entire network.

Reply:
It now does slice random, https://docs.rs/rand/0.6.5/rand/seq/trait.SliceRandom.html#tymethod.choose_multiple no need to shuffle